File: 알고리즘/2/Reverse Vowels of a String.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = 'aeiouAEIOU'
        basic = [ele for ele in s if ele in vowels]
        basic.reverse()        
        i=0
        result =''
        for ele in s:
            if ele not in vowels:
                result+='ele'
            else:
                result+=basic[i]
                i+=1
        return result
        
        
        
        
        
        

File: algorithms/2/Reverse Vowels of a String.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string reverseVowels(string s) {
        string vowels = "aeiouAEIOU";
        string search ="";
        for(auto ele: s){
            if(vowels.find(ele)){
                search+=ele;
            }
        }
        reverse(search.begin(), search.end());
        
        string result = "";
        int i=0;
        for (auto ele : s){
            if(vowels.find(ele) == string::npos){
                result+=ele;
            } else{
                result += search[i];
                i+=1;
            }
        }

        return result;
    }
};